### ISSUE TITLE
Notifications that are not delivered inmediately to webView could be lost.

### ISSUE THREAD

**Issue by nicoabie** [phonegap/phonegap-plugin-push#475]
I'm looking for ways to overcome this issue that has two ways of appearing:

1. there is a cached notification and a new one arrives
2. there is a cached notification and the user restarts the device

I'm considering [NSCoding protocol](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Protocols/NSCoding_Protocol/) for the IOS platform in the temp folder as it does not requires an any 3rd party library.

Android platform still needs investigation.

@macdonst, @fredgalvao thoughts?

**Comment by macdonst** [comment]
@nicoabie expand the cache into an map instead of just storing the latest message?

**Comment by nicoabie** [comment]
Yes @macdonst , that just solves number 1. If the user restarts the device the in memory map will be lost. Trying to kill to birds with one shot.

**Comment by macdonst** [comment]
@nicoabie I think you might be over engineering this solution. Notifications just aren't that important :smiley: 

For important notifications that have to be acted on then send a background notification that will be processed immediately.

**Comment by nicoabie** [comment]
Hahaha in our use case they are @macdonst!
We are sending notifications with content available and we need to ensure that [this](https://github.com/phonegap/phonegap-plugin-push/blob/master/src/android/com/adobe/phonegap/push/GCMIntentService.java#L215) effectively calls our on notification js callback. 

We need to display a popup in the app the the data we are sending so if that notification is lost because it was on memory there is no way we can fulfill that requirement. :S
